uaa/src/main/webapp/WEB-INF/spring/saml-providers.xml

```diff
@@ -144,6 +149,8 @@
         <bean id="samlWebSSOProcessingFilter" class="org.springframework.security.saml.SAMLProcessingFilter">
             <property name="authenticationManager" ref="authenticationManager" />
             <property name="authenticationSuccessHandler" ref="successRedirectHandler" />
+            <property name="authenticationFailureHandler" ref="samlLoginFailureHandler" />
+            <property name="contextProvider" ref="contextProvider" />
```

Comment:
Thanks for adding this, you may notice that it gets set, even without it, because of the @Autowired annotation. 

    @Autowired
    public void setContextProvider(SAMLContextProvider contextProvider) {
        Assert.notNull(contextProvider, "Context provider can't be null");
        this.contextProvider = contextProvider;
    }